### ISSUE TITLE
Bucket creation errors out with bucket_missing_file_lock but actually creates the bucket 

### ISSUE THREAD

**Issue by brandonrjacobs** [Backblaze/terraform-provider-b2#57]
```terraform
$ terraform apply -input=false "planfile"
module.backblaze.b2_bucket.backups["backups"]: Creating...

╷
│ Error: dateutil/zoneinfo/__init__.py:26: UserWarning: I/O error(2): No such file or directory
│ b2sdk/exception.py:606: UserWarning: bad request exception with an unknown `code`. message=The bucket is not file lock enabled, code=bucket_missing_file_lock
│ Traceback (most recent call last):
│   File "b2_terraform/provider_tool.py", line 515, in run_command
│   File "b2_terraform/provider_tool.py", line 131, in run
│   File "b2_terraform/provider_tool.py", line 239, in resource_create
│   File "logfury/_logfury/trace_call.py", line 86, in wrapper
│   File "b2sdk/bucket.py", line 171, in update
│   File "b2sdk/session.py", line 325, in update_bucket
│   File "b2sdk/session.py", line 445, in _wrap_default_token
│   File "b2sdk/session.py", line 451, in _wrap_token
│   File "b2sdk/session.py", line 466, in _reauthorization_loop
│   File "b2sdk/session.py", line 456, in _api_token_callback
│   File "b2sdk/raw_api.py", line 771, in update_bucket
│   File "b2sdk/raw_api.py", line 413, in _post_json
│   File "b2sdk/b2http.py", line 2[76](https://gitlab.com/coreweave/terraform/-/jobs/5571088508#L76), in post_json_return_json
│   File "b2sdk/b2http.py", line 235, in post_content_return_json
│   File "b2sdk/b2http.py", line 462, in _translate_and_retry
│   File "b2sdk/b2http.py", line 397, in _translate_errors
│ b2sdk.exception.BadRequest: The bucket is not file lock enabled (bucket_missing_file_lock)
│ 
│ 
│   with module.backblaze.b2_bucket.backups["backups"],
│   on ../../modules/backblaze/backblaze.tf line 22, in resource "b2_bucket" "backups":
│   22: resource "b2_bucket" "etcd-backups" {
```

Here is the resource that was planned:
```terraform
# module.backblaze.b2_bucket.etcd-backups["rno2-int-etcd-backups"] will be created
  + resource "b2_bucket" "backups" {
      + account_id  = (known after apply)
      + bucket_id   = (known after apply)
      + bucket_name = "backups"
      + bucket_type = "allPrivate"
      + id          = (known after apply)
      + options     = (known after apply)
      + revision    = (known after apply)
      + default_server_side_encryption {
          + algorithm = "AES256"
          + mode      = "SSE-B2"
        }
      + file_lock_configuration {
          + is_file_lock_enabled = false
          + default_retention {
              + mode = "governance"
              + period {
                  + duration = 365
                  + unit     = "days"
                }
            }
        }
    }
```

We are using the latest terraform provider for backblaze/b2 and using Terraform 1.5.6.

We also see the bucket is created in Backblaze in our console but it does not show it as created and terraform knows nothing of it's state.

**Comment by ppolewicz** [comment]
You can't configure retention unless you enable file lock first.

Change `is_file_lock_enabled` from `false` to `true`.

**Comment by brandonrjacobs** [comment]
I also wouldn't expect it to create the bucket if it errored like this, or if it did it should at least update the state to know about the created bucket so that subsequent plans don't show it needing to be created still.

**Comment by ppolewicz** [comment]
I think the bucket has been created without file lock being enabled on it (due to a bug) and now for some reason enabling file lock doesn't work after the bucket has been created. The reason might be that enabling file lock on an existing bucket has been added fairly recently, to the CLI on 2022-09-20, and the last release of the terraform provider was in March.

For now as a workaround you can either use B2 CLI to enable file lock on that bucket, or you can remove the bucket (assuming it's empty) from the web panel and then run terraform again.

For a permanent fix we'll release a new version of the terraform provider and test to make sure upgrading a bucket to an object-lock-capable works as intended.

**Comment by vbaltrusaitis-reef** [comment]
Issue fixed, version `v0.8.5` released.
